domains/certificates/Query.API/API/GranularCertificateIssuer/EnergyMeasuredConsumer.cs

```diff
@@ -26,28 +28,35 @@ public async Task Consume(ConsumeContext<EnergyMeasuredIntegrationEvent> context
     {
         var message = context.Message;
 
-        var contract = await contractService.GetByGSRN(message.GSRN, context.CancellationToken);
+        var contracts = await contractService.GetByGSRN(message.GSRN, context.CancellationToken);
 
-        if (!ShouldEventBeProduced(contract, message))
+        if (contracts.IsEmpty())
         {
-            logger.LogInformation("No production certificate created for {message}", message);
             return;
         }
 
-        var productionCertificate = new ProductionCertificate(
-            contract!.GridArea,
-            new Period(message.DateFrom, message.DateTo),
-            new Technology(FuelCode: "F00000000", TechCode: "T070000"),
-            contract.MeteringPointOwner,
-            message.GSRN,
-            message.Quantity);
+        foreach (var contract in contracts)
+        {
+            if (!ShouldEventBeProduced(contract, message))
+            {
+                logger.LogInformation("No production certificate created for {Message}", message);
```

Comment:
This log statement can be falsely written to the log if there are multiple contracts in `contracts` and the last one is a hit. This log statement should only be printed if there really is no production certificate being created.

Please be aware that we have a Grafana dashboard that is dependent on these log statements.